crud.go

```diff
@@ -387,12 +387,43 @@ func (bucket *WalrusBucket) GetXattr(k string, xattrKey string, xv interface{})
 	return 0, errors.New("GetXattr not implemented for walrus")
 }
 
-func (bucket *WalrusBucket) GetSubDocRaw(k string, subdocKey string)(value []byte, casOut uint64, err error){
-	return nil, 0, errors.New("GetSubDocRaw not implemented for walrus")
+// GetSubDocRaw Walrus implementation only works with a top-level subdocKey
+func (bucket *WalrusBucket) GetSubDocRaw(k string, subdocKey string) (value []byte, casOut uint64, err error) {
+	var fullDoc map[string]interface{}
+	casOut, err = bucket.Get(k, &fullDoc)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	if fullDoc[subdocKey] == nil {
```

Comment:
Check for presence of the key, rather than the value, otherwise a doc like this will return an error.

```json
{"foo":null}
```

I'd expect `GetSubDocRaw(doc, "foo")` to return `{"foo":null}, casOut, nil` 

```suggestion
	if _, ok := fullDoc[subdocKey]; !ok {
```